### PR TITLE
Enhancements to support recovery plan jobs.

### PR DESCRIPTION
**Pull request by prakash-nutanix**
RecoveryPlanJobs are the way to run recovery plans during failover and failback. This patch adds support for all the API endpoints related to recovery plan jobs.

### CODE REVIEW

File: v3/v3_service.go [hunk 1]
```diff
@@ -2282,3 +2305,76 @@ func (op Operations) UpdateAddressGroup(ctx context.Context, uuid string, body *
 
 	return op.client.Do(ctx, req, nil)
 }
+
+func (op Operations) CreateRecoveryPlanJob(ctx context.Context, request *RecoveryPlanJobIntentInput) (*RecoveryPlanJobResponse, error) {
```
Comment (deepakm-ntnx) [inline v3/v3_service.go:2309]: comment header required to be consistent with other functions

File: v3/v3_structs.go [hunk 2]
```diff
@@ -1625,10 +1625,9 @@ type Metadata struct {
 	// Applied on Prism Central only. Indicate whether force to translate the spec of the fanout request to fit the target cluster API schema.
 	ShouldForceTranslate *bool `json:"should_force_translate,omitempty" mapstructure:"should_force_translate,omitempty"`
 
-	// TODO: add if necessary
-	// CategoriesMapping    map[string][]string `json:"categories_mapping,omitempty" mapstructure:"categories_mapping,omitempty"`
-	// EntityVersion        *string             `json:"entity_version,omitempty" mapstructure:"entity_version,omitempty"`
-	// UseCategoriesMapping *bool               `json:"use_categories_mapping,omitempty" mapstructure:"use_categories_mapping,omitempty"`
+	CategoriesMapping    map[string][]string `json:"categories_mapping,omitempty" mapstructure:"categories_mapping,omitempty"`
+	EntityVersion        *string             `json:"entity_version,omitempty" mapstructure:"entity_version,omitempty"`
+	UseCategoriesMapping *bool               `json:"use_categories_mapping,omitempty" mapstructure:"use_categories_mapping,omitempty"`
```
Comment (deepakm-ntnx) [inline v3/v3_structs.go:1630]: where is this used?

File: v3/v3_service.go (hunk 1 above)
Comment (prakash-nutanix) [reply v3/v3_service.go:2314]: Added to all methods newly added.

File: v3/v3_structs.go (hunk 2 above)
Comment (prakash-nutanix) [reply v3/v3_structs.go:1630]: This is a part of of standard metadata returned by most PC v3 API calls.

**Review by deepakm-ntnx — DISMISSED** [review]

**Comment by thunderboltsid** [comment]
Can you add entries to the CHANGELOG.md?

**Comment by deepakm-ntnx** [comment]
> Can you add entries to the CHANGELOG.md?

@thunderboltsid lets explore options to make this automated

**Comment by thunderboltsid** [comment]
> > Can you add entries to the CHANGELOG.md?
> 
> @thunderboltsid lets explore options to make this automated

If everyone pledges to write conventional commits, sure we can. Until then, definitely no. http://conventionalcommits.org

**Review by thunderboltsid — APPROVED** [review]

**Review by deepakm-ntnx — APPROVED** [review]

**Pull request merged**